wfc_dylib/src/lib.rs

```diff
@@ -452,11 +477,23 @@ pub extern "C" fn wfc_observe(
         &mut *wfc_rng_state_handle.0
     };
 
+    let observations = unsafe {
+        assert!(!spent_observations.is_null());
+        &mut *spent_observations
+    };
+
+    *observations = 0;
```

Comment:
This is a minor issue, but for readability and clarity, it is a good practice to only assign to the pointer once. I'd have a local variable for the observation count and assign it to the pointer just before the function returns (in either of the branches it returns). This is the same concept as doing IO on the periphery of the function and pure computation inside. View writing to the pointer as IO.

There might also be a small speed penalty for doing it the way you do it, but maybe the compiler dealt with that.

Reply:
Let me know if you like it better. Couldn't come up with a nicer solution.